tests/contract/pyproject.toml

```
@@ -7,18 +7,18 @@ license = "Mozilla Public License Version 2.0"
 readme = "README.md"
 
 [tool.poetry.dependencies]
-python = "^3.11"
+python = "^3.12"
 requests = "^2.31.0"
 
 [tool.poetry.group.client.dependencies]
-pydantic = "^2.1.0"
-pytest = "^7.3.1"
-PyYAML = "^6.0"
+pydantic = "^2.4.2"
+pytest = "^7.4.3"
+PyYAML = "6.0.1"
```

I also have the `Installing pyyaml (6.0.1): Failed` error on my local at the moment.


```suggestion
PyYAML = "^6.0.1"
```